### PR TITLE
Add missing `git` icons

### PR DESCRIPTION
**Pull request by danielbayley**
* Added `git-commit-vertical` (and subsequently renamed `git-commit`[`-horizontal`])
* Added `git-pull-request-create`
* Refined `git-fork` to better match [`network`](https://lucide.dev/icons/network)
* Refined `git-pull-request-closed` to better match other `*-x` icons (as most `x`es are 5x5 px, not 6x6)
* Added `git-`[`pull-request-`[`create`/`compare`]`-arrow`/`s` variants, only making them slightly wider for better consistency than just scaling them up on both axes.
* Added `git-graph` icon (maybe should be `git-commit-graph`?) like in VS Code…

Untitled
<br/>
<br/>

Full preview of changes in https://github.com/lucide-icons/lucide/pull/995#issuecomment-1509285280.

### CODE REVIEW

**Comment by karsa-mistmere** [comment]
I'm afraid the bookmark and the tiny arrows do not have sufficient padding (a minimum of 2px spacing would be required by our guidelines).

**Comment by karsa-mistmere** [comment]
These latest icons should probably use some sort of name along the lines of `inbox-{...}`, but they also have some guideline violations I'm afraid:
image

**Comment by danielbayley** [comment]
> I'm afraid the bookmark and the tiny arrows do not have sufficient padding (a minimum of 2px spacing would be required by our guidelines).

Tiny arrows, which do you mean? One of them was misaligned by mistake, fixed now…

**Comment by karsa-mistmere** [comment]
> Tiny arrows, which do you mean? One of them was misaligned by mistake, fixed now…

E.g. the spacing marked with yellow here: 
image

**Comment by danielbayley** [comment]
Other alternative would be:

Artboard 1@30x

**Comment by danielbayley** [comment]
Or, maybe:

inbox-2

**Comment by danielbayley** [comment]
Or, maybe:

inbox-2

**Comment by danielbayley** [comment]
Or, maybe:

inbox-2

**Comment by danielbayley** [comment]
@karsa-mistmere Fixed the repo icons…

repos

**Review by ericfennis — COMMENTED** [review]
Thanks for these new icons. So next time try to split up your icons with multiple PRs grouped visually. This makes reviewing icon PRs a lot easier. 

For this PR:
I agree with @karsa-mistmere the arrows are a bit too close to the line. If you preview the icon on 100% it looks a bit messy. Not sure how you can fix this. Will try to help you with that one.

For the `git-repo-*` icons: 
Would be nice if the key and the lock are visually the same as the ones used in the `files` icons. And I'm not sure about the naming. The icons can be used for other things as `git-repos` as well so maybe we can give them more generic names.

And the `git-stash` icons should be named `inbox-in` and `inbox-out`. You may better add them to your other PR #998, I think they belong there.

**Comment by danielbayley** [comment]
>  So next time try to split up your icons with multiple PRs grouped visually. This makes reviewing icon PRs a lot easier.

Yeah, I intended to… Just bundled all the `git` related ones together for now, I’ll break them down more now there is a better consensus…

I’ll move the repo icons to a separate PR, as I am happy with the versions I’ve tweaked offline… and I’ll leave this one open for the `git` command icons, as the arrows on those are trickier to solve.

> I'm not sure about the naming. The icons can be used for other things as `git-repos` as well so maybe we can give them more generic names.

Something like `book-marked`?

> And the `git-stash` icons should be named `inbox-in` and `inbox-out`. You may better add them to your other PR https://github.com/lucide-icons/lucide/pull/998, I think they belong there.

Agreed…

File: icons/git-pull-request-2.svg [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<svg
```
Comment (ericfennis) [inline icons/git-pull-request-2.svg:1]: I'm not pleased with the current size of the arrow from this one. 

I tried to give one a shot. But It's hard. I came up with two alternatives, but they have some tradeoffs..

Comment (ericfennis) [reply icons/git-pull-request-2.svg:1]: Forgot to post the screenshot. Let me know what you think
Screenshot 2023-04-10 at 11 28 12

Comment (danielbayley) [reply icons/git-pull-request-2.svg:1]: @ericfennis I think alt 1…

Comment (ericfennis) [reply icons/git-pull-request-2.svg:1]: What do you think @karsa-mistmere ?

Comment (karsa-mistmere) [reply icons/git-pull-request-2.svg:1]: Honestly, I guess I'm mostly in favour of keeping just the ones with no arrows. 🤷‍♂️

To me, both alternatives feel forced; Alt 1 will make our git icons inconsistent and Alt 2 doesn't quite stick to the guidelines. 🙈

Comment (danielbayley) [reply icons/git-pull-request-2.svg:1]: > Alt 1 will make our git icons inconsistent

Although they could easily be made consistent… Also visually centred and proportionally more square:

Untitled

**Comment by ericfennis** [comment]
Still not sure about breaking the consistency with other git accounts. But the arrows are better now.
I keep this one open, I want to know what others think of this.

**Comment by danielbayley** [comment]
> Still not sure about breaking the consistency with other git accounts.

I don’t think they should be inconsistent either, but no reason why they have to be… For reference:

Untitled

Top row are additions to the existing set, with the same proportions. Row 2 is the existing set. Row 3 are adapted to keep consistency with row 4, which are the arrow variations.

Thoughts? @ericfennis @karsa-mistmere @jguddas

**Comment by danielbayley** [comment]
> can you provide these in a Figma file first so that I can play around with them a bit? :)

@karsa-mistmere https://figma.com/file/LjycnaUj6vyISowz39qNfX/lucide%2Fgit-icons?node-id=0%3A1&t=uGJ176Kfok9Y3qfg-1

**Comment by danielbayley** [comment]
See latest update… @karsa-mistmere @ericfennis @jguddas

**Review by ericfennis — COMMENTED** [review]
Ok In general some looks not that bad taking a bit more space.
I agree with @karsa-mistmere the `git-pull-request-closed` and `git-pull-request-draft` looking not really balanced. 

For me, I prefer to keep the old designs for `git-compare` and `git-pull-request` and add your new designs as alternative icons. So under a new name or add this as as second variant.

**Comment by danielbayley** [comment]
> For me, I prefer to keep the old designs for `git-compare` and `git-pull-request` and add your new designs as alternative icons. So under a new name or add this as as second variant.

Ok, so I `revert`ed scaling them all up, and instead implemented the following summary of changes…

* Added `git-commit-vertical` (and subsequently renamed `git-commit`[`-horizontal`])
* Added `git-pull-request-create`
* Refined `git-fork` to better match [`network`](https://lucide.dev/icons/network)
* Refined `git-pull-request-closed` to better match other `*-x` icons (as most `x`es are 5x5 px, not 6x6)
* Added `git-`[`pull-request-`[`create`/`compare`]`-arrow`/`s` variants, only making them slightly wider for better consistency than just scaling them up on both axes.
* Added `git-graph` icon as outlined [above](https://github.com/lucide-icons/lucide/pull/995#issue-1644855557)

@ericfennis See latest https://github.com/lucide-icons/lucide/pull/995#issuecomment-1509285280.

cc: @karsa-mistmere @jguddas

File: icons/git-pull-request-closed.svg [hunk 2]
```diff
@@ -12,7 +12,7 @@
   <circle cx="18" cy="18" r="3" />
   <circle cx="6" cy="6" r="3" />
   <path d="M18 11.5V15" />
-  <path d="m21 3-6 6" />
-  <path d="m21 9-6-6" />
+  <path d="m20.5 3.5-5 5" />
```
[the remark this answers is not in the document]
Comment (danielbayley) [reply icons/git-pull-request-closed.svg]: @ericfennis Fixed.

**Review by ericfennis — APPROVED** [review]

**Pull request merged**